src/uct/api/uct.h

```diff
@@ -1457,6 +1459,14 @@ typedef struct uct_md_mem_attr {
      * to uct_md_mem_query is returned as is.
      */
     size_t            alloc_length;
+
+    /**
+     * fd object for the provided buffer region.
+     * If the md does not support querying fd object associated with the region,
+     * then -1 is returned. It is responsibility of the user to close returned
+     * fd once used.
+     */
+    int               dmabuf_fd;
```

Comment:
We really need to document here what is DMAbuf . Since DMAbuf is a very specific Linux kernel feature, it would be a good to provide references to linux documentation, examples, etc.

Reply:
@shamisp wanted to check if the reference added suffices or not. Lmk when possible.

Reply:
@Akshay-Venkatesh can you please confirm that it generates URL link in Doxygen ? According to Doxygen documentation it supposed to do this automatically but sometimes it gets broken.

Reply:
@shamisp `make docs` and the following seems to suggest that URL link is being generated. 
```
$ grep -r "linuxgraphics" docs/doxygen-doc/
docs/doxygen-doc/man/man3/UCT_MD.3:int \fIfd\fP A file descriptor to expose memory regions across devices\&. One example could be a dmabuf file des
criptor for the provided buffer region\&. Refer (https://01.org/linuxgraphics/gfx-docs/drm/driver-api/dma-buf.html)\&. If the md does not support q
uerying fd object associated with the region, then UCT_FD_INVALID is returned\&. It is responsibility of the user to close returned fd once used\&.
 
docs/doxygen-doc/html/group___u_c_t___m_d.html:<p>A file descriptor to expose memory regions across devices. One example could be a dmabuf file des
criptor for the provided buffer region. Refer (<a href="https://01.org/linuxgraphics/gfx-docs/drm/driver-api/dma-buf.html">https://01.org/linuxgrap
hics/gfx-docs/drm/driver-api/dma-buf.html</a>). If the md does not support querying fd object associated with the region, then UCT_FD_INVALID is re
turned. It is responsibility of the user to close returned fd once used. </p>
```